addons/payment/models/payment_method.py

```diff
@@ -294,6 +299,25 @@ def _get_compatible_payment_methods(
                 reason=REPORT_REASONS_MAPPING['express_checkout_not_supported'],
             )
 
+        # Filter pms that does not support manual capture.
+        unfiltered_pms = payment_methods
+        automatic_capture_pms = payment_methods.filtered(
+            lambda p: not p.support_manual_capture
+        )
+        providers = self.env['payment.provider'].browse(provider_ids)
+        capture_manually_active_pp = providers.filtered('capture_manually')
+        if automatic_capture_pms and capture_manually_active_pp:
+            for pm in automatic_capture_pms:
+                if any(p in pm.provider_ids for p in capture_manually_active_pp):
+                    payment_methods -= pm
+
+        payment_utils.add_to_report(
+            report,
+            unfiltered_pms - payment_methods,
+            available=False,
+            reason=REPORT_REASONS_MAPPING['manual_capture_not_supported'],
+        )
```

Comment:
I don't think we should do that at all. Manual capture is an action that takes place *after* payment, not *during* payment.

Instead, we could remove the filter and add a constraint that prevents enabling manual capture on a provider that has enabled some PMs that don't support manual capture. This is equivalent to filtering since it prevents using the PMs and requires disabling them first.